### PR TITLE
Issue library QA index 

### PR DESCRIPTION
**Pull request by aapomm**
### Summary

Implement the following changes:

- Datatable state JS cleanup and update bootstrap class
- Remove QA JSON endpoint response

### Check List

~- [ ] Added a CHANGELOG entry~

### CODE REVIEW

File: app/controllers/qa/issues_controller.rb
Comment (etdsoft) [inline app/controllers/qa/issues_controller.rb:1]: Why are these changes required?

Comment (aapomm) [reply app/controllers/qa/issues_controller.rb:1]: Removed as it is unused

File: app/assets/javascripts/shared/datatable/state.js [hunk 1]
```diff
@@ -1,53 +1,61 @@
-$(document).on('preInit.dt', 'body.issues.index', function (e, settings) {
-  var $paths = $('[data-behavior~=datatable-paths]');
-  if ($paths.data('table-state-url') == undefined) {
-    return;
-  }
+$(document).on(
+  'preInit.dt',
+  'body.issues.index',
```
Comment (etdsoft) [inline app/assets/javascripts/shared/datatable/state.js:3]: Is this code in the right location? If it's tied to a page, then it shouldn't live under /shared/datable/ but under tylium/pages/

If this is really shared code across multiple DTs then we need to remove the hard-coding and use a data-behavior or some other trigger to hook into it.

File: app/assets/javascripts/shared/datatable/state.js [hunk 2]
```diff
@@ -1,53 +1,61 @@
...
+  }
+);
 
 DradisDatatable.prototype.setupStateButtons = function () {
   var states = $('[data-behavior~=dradis-datatable]').data('state-icons'),
     stateButtons = [],
     api = this;
 
-  if ($('[data-behavior~=qa-viewer]').length > 0) {
+  if ($('[data-behavior~=qa-viewer]').length) {
```
Comment (etdsoft) [inline app/assets/javascripts/shared/datatable/state.js:51]: same sort of thing. Doesn't belong hard-coded in a shared/ module.

File: app/controllers/qa/issues_controller.rb [hunk 3]
```diff
@@ -46,11 +46,8 @@ def multiple_update
             redirect_to project_issues_path(current_project), notice: 'State updated successfully.'
           end
         end
-
-        format.json { head :ok }
```
Comment (etdsoft) [inline app/controllers/qa/issues_controller.rb:50]: did you do a sweep to see anywhere else where we can remove unnecessary `format.json`s?

File: app/assets/javascripts/shared/datatable/state.js (hunk 1 above)
Comment (aapomm) [reply app/assets/javascripts/shared/datatable/state.js:3]: Apparently the hardcoding is not needed considering that we're already testing for the presence of the `state-url`

File: app/assets/javascripts/shared/datatable/state.js (hunk 2 above)
Comment (aapomm) [reply app/assets/javascripts/shared/datatable/state.js]: Moved the state icon handling to the view ✅

File: app/controllers/qa/issues_controller.rb (hunk 3 above)
Comment (aapomm) [reply app/controllers/qa/issues_controller.rb:50]: It doesn't look like there's other unused format.jsons

File: app/assets/javascripts/shared/datatable/state.js [hunk 4]
```diff
@@ -72,7 +74,7 @@ DradisDatatable.prototype.updateRecordState = function (newState) {
       data: {
         ids: DradisDatatable.prototype.rowIds(selectedRows),
         state: newState,
-        return_to: $('[data-behavior~=qa-viewer]').length > 0 ? 'qa' : null
+        return_to: $('[data-behavior~=qa-viewer]').length > 0 ? 'qa' : null,
```
Comment (etdsoft) [inline app/assets/javascripts/shared/datatable/state.js:77]: does this need to go?

Comment (aapomm) [reply app/assets/javascripts/shared/datatable/state.js:77]: Replaced with a data attribute instead ✅
